src/Components/test/testassets/BasicTestApp/CloseEventComponent.razor

```diff
@@ -0,0 +1,25 @@
+﻿<div>
+    <p>
+        Output: <span id="output">@message</span>
+    </p>
+
+
+    <p>Close the dialog.</p>
+
+    <dialog @onclose="OnClose" open>
+        <form method="dialog">
+            <button id="dialog-close">Close</button>
+        </form>
+    </dialog>
+
+</div>
+
+@code {
+    string message { get; set; }
+
+    void OnClose(EventArgs e)
+    {
+        message += "onclose,";
+        StateHasChanged();
```

Comment:
This call shouldn't be needed, since the event dispatcher causes the equivalent to `StateHasChanged` implicitly anyway. Would be good to update this file so that the tests validate it works without explicitly calling this.

Reply:
I'm just going to remove it now, because if it works and the tests pass, we can still merge this in time for .NET 8. If it misses the end-of-day deadline today, it will slip into .NET 9!

Reply:
I just copied it from the `ToggleEventComponent` maybe there are more unnecessary `StateHasChanged` in other test components as well.